pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -906,7 +906,30 @@ func (r *ReconcilePerconaServerMongoDB) reconcileMongos(cr *api.PerconaServerMon
 		return errors.Wrap(err, "check if mongos custom configuration exists")
 	}
 
-	deplSpec, err := psmdb.MongosDeploymentSpec(cr, opPod, log, customConfig)
+	cfgPods, err := r.getRSPods(cr, api.ConfigReplSetName)
+	if err != nil {
+		return errors.Wrap(err, "get configsvr pods")
+	}
+
+	// wait all configsvr pods to prevent unnecessary updates to mongos deployment
+	if int(cr.Spec.Sharding.ConfigsvrReplSet.Size) > len(cfgPods.Items) {
+		return nil
+	}
+
+	cfgInstances := make([]string, 0, len(cfgPods.Items))
```

Comment:
we already know the size of `cr.Spec.Sharding.ConfigsvrReplSet.ExternalNodes`, so we can put it to `make` function too

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/729/commits/db744b631bde51f6969c4340d944ef36d2ce2b10